### DISCUSSION TITLE
feature: being able to call variables in dialogue choices

### DISCUSSION THREAD

**Discussion by dhmello** [AscensionGameDev/Intersect-Engine#1997] [Ideas]
### Description

It would be very good and would make quests and event systems more complete by being able to call a variable in choices in dialogue with an NPC or system.

Where during a player-by-player variable, depending on what they have already done in another quest, there will be a different answer available to give to the NPC or event

### Intended Use-Case

Where during a player-by-player variable, depending on what they have already done in another quest, there will be a different answer available to give to the NPC or event

### Duplicate Check

- [x] This feature request is not a duplicate to the best of my knowledge.

**Comment by AlefMichel** [comment]
This can enrich the player's experience and create a more immersive feeling that the player's actions have real consequences in the game world.

The ability to use variables to track the player's progress and adapt interactions based on previous decisions is a common technique in game design and interactive storytelling. This allows players to feel more involved in the story because their choices have a visible impact on the game

**Comment by Richy1111** [comment]
Isn't this already a thing?
Ironically enough, we can even use the \pv{ID} inside the answer boxes too. I never even knew that. :P
But yeah, as for variables, can use them in option texts too, and you don't even need to randomize like I did, you can just conditional any variable that flips/changes elsewhere if wanted and can change dialogue string variables at any time as well.

![D's](https://github.com/AscensionGameDev/Intersect-Engine/assets/72468758/62c6e60a-5290-4bcf-a704-33ecde4e3425)
![Event D's](https://github.com/AscensionGameDev/Intersect-Engine/assets/72468758/794cfdc3-dc2b-493f-9897-507626879ea6)
![Event In Play](https://github.com/AscensionGameDev/Intersect-Engine/assets/72468758/f5fafdc8-551e-483c-9a02-49a69d47c5fd)
![Event In Play 2](https://github.com/AscensionGameDev/Intersect-Engine/assets/72468758/0d536caa-1a59-44b9-b2a1-7a928a757441)
![Option String Calls](https://github.com/AscensionGameDev/Intersect-Engine/assets/72468758/c888c165-14ff-411e-b2e2-01d905a3b18e)

**Comment by dhmello** [comment]
Yes, but look at how many conditions you had to create just to change the response variable....

Now imagine giving an answer, "YES DO THE QUESTION FOR \pv{times} TIMES"
Another answer: Yes, pay \pv{value} for the service
Another answer: Yes, deposit \pv{value} in the bank

This can be changed with texts too, in addition to int or bool...

**Reply by Richy1111** [reply]
You still need to control/change the variables regardless. So you need atleast 5 string variables (4 changeable answers) and (1 changeable text). Then you'd need int and bool variables that determine changes for any given time. The story lines aren't going to build itself as we'd still need to event it all in.

I took the harder road and made a randomized dialogue, this doesn't need to be the case except the fact that conditions still need to be met to change anything anyway.

I don't think there really is any other way to improve it? Even if there was a variable string editor, you'd still need to add/change all the strings and you'd still need to call them inside the option text command. So I really dunno how a coded feature could benefit?

**Comment by dhmello** [comment]
I'm going crazy, this is now possible to do hehe
Sorry about the mess!
